### DISCUSSION TITLE
AWS issues?

### DISCUSSION THREAD

**Discussion by jordyperlee** [timdorr/tesla-api#339] [Q&A]
Hey all,

So my code (based on TeslaJS) works perfectly fine on my machine, but on AWS the step 1 call seems to always time out/get blocked. Has anyone else experienced something similar?

**Comment by labdel** [comment]
Hi there,
yes, I experinced the same with my EC2 machine. Unfortunately I had to work with a network engineer to route traffic through a VPN so it does not arrive from AWS.

Regards

**Reply by mehdirahbe** [reply]
I have same problem, people cannot refresh their token on my heroku (AWS based) site anymore.  Which VPN did you use? How should I configure my python program to use it?

**Reply by labdel** [reply]
I do not know about networking, a network engineer created someting in AWS called VPN and routed traffic to tesla into somewhere else and then into tesla. I am not too sure how He did it, I could give his contact and to work with him though.
